util/font-loader.js

```diff
@@ -0,0 +1,30 @@
+import FontFaceObserver from 'fontfaceobserver'
+
+(async () => {
+  if (sessionStorage.fontsLoaded) {
+    document.documentElement.classList.add('font-loaded')
+    return
+  }
+
+  const lato = new FontFaceObserver('Lato')
+  const latoBold = new FontFaceObserver('Lato', {
+    weight: 700
+  })
+  const latoItalic = new FontFaceObserver('Lato', {
+    style: 'italic'
+  })
+  const latoBoldItalic = new FontFaceObserver('Lato', {
+    weight: 700,
+    style: 'italic'
+  })
+
+  await Promise.all([
```

Comment:
What I used to do is more something like that:

```
(function () {

  // Optimization for Repeat Views
  if (sessionStorage.criticalFoftDataUriFontsLoaded) {
      document.documentElement.classList.add('fonts-critical', 'fonts-loaded')
      return
  }
  const fontASubset = new FontFaceObserver('LatoSubset')
  fontASubset.load().then(_ => {
    document.documentElement.classList.add('fonts-critical')

    const fontA = new FontFaceObserver('Lato')
    const fontB = new FontFaceObserver('LatoBold', {weight: 700})
    const fontC = new FontFaceObserver('LatoItalic', {style: 'italic'})
    const fontD = new FontFaceObserver('LatoBoldItalic', {weight: 700, style: 'italic'})
    Promise.all([fontA.load(), fontB.load(), fontC.load(), fontD.load()]).then(_ => {
      document.documentElement.classList.add('fonts-loaded')

      // Optimization for Repeat Views
      sessionStorage.criticalFoftDataUriFontsLoaded = true
    })
  })
})()
```

This way you start to render the most critical font first, in our (French) case you can just remove the LatoSubset and use Lato instead. Tell me if it's not clear we can setup a 10min visio about that.

Reply:
Except for the different font names (what’s the point?), how is this different from what I did?

And yeah, an ASCII subset is not necessary.

Reply:
The proposed version is a two-steps process.